### DISCUSSION TITLE
the `remote` flag in the `list` command

### DISCUSSION THREAD

**Discussion by tareknaser** [coffee-tools/coffee#129] [Ideas]
Previously, we had the `remotes` flag output all possible plugins available to be installed.
The output JSON looked like this.
![image](https://github.com/coffee-tools/coffee/assets/60650661/d809e9ce-3b2d-4718-ab26-faef5dbfb718)
The first object was the installed plugins and the second object was the plugins available for installation categorized by the remote repository.
This was then changed to only showing the name of the remote repositories available locally

```
● Plugin installed
╭─────────────────────────────────╮
│ ●   Language   Name   Exec path │
├─────────────────────────────────┤
╰─────────────────────────────────╯

● List of repositories
╭───────────────────────────────────────────────────────────────────────────╮
│ ●   Repository Alias   URL                                     N. Plugins │
├───────────────────────────────────────────────────────────────────────────┤
│ ●   lightningd         https://github.com/lightningd/plugins   27         │
╰───────────────────────────────────────────────────────────────────────────╯
```
but then it was deprecated because this is identical to `remote --list` which makes more sense given this output.

Now, we are considering the possibility of reintroducing the original implementation of the remotes flag. However, before proceeding, we need to discuss the optimal format of the output.

**Comment by vincenzopalazzo** [comment]
This is a good point, in my humble opinion. I think that a good design would be the following:

- `remote list`: shows the list of the repository, and in this case, it would also be nice to list all the plugins inside it. I should play with it a little bit to see if we can have a sub-item.
- `list` (better rename this to `ls`?): returns all the installed plugins with an additional column that indicates in which repository the plugin is contained.

**Comment by vincenzopalazzo** [comment]
I this this is implemented
